### PR TITLE
[zh-cn]: revise the translation of images in html

### PR DESCRIPTION
**Pull request by hanyujie2002**
Relates to #18313

### CODE REVIEW

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 1]
```diff
@@ -165,145 +206,206 @@ slug: Learn/HTML/Multimedia_and_embedding/Images_in_HTML
...
-    </script>
-  </body>
-</html>
+<h2>实时输出</h2>
+
+<div class="output" style="min-height: 50px;"></div>
+
+<h2>可编辑代码</h2>
+<p class="a11y-label">
+  按 ESC 退出编辑区域（按 Tab 可插入制表符）.
```
Comment (hanyujie2002) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:226]:
```suggestion
  按 ESC 退出编辑区域（按 Tab 可插入制表符）
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 2]
```diff
@@ -1,161 +1,202 @@
...
       </td>
     </tr>
   </tbody>
 </table>
 
-## 怎样将一幅图片放到网页上？
+## 怎样将图片放到网页上？
 
-我们可以用 {{htmlelement("img")}} 元素来把图片放到网页上。它是一个空元素（它不需要包含文本内容或闭合标签），最少只需要一个 `src` （一般读作其全称 _source_）来使其生效。`src` 属性包含了指向我们想要引入的图片的路径，可以是相对路径或绝对 URL，就像 {{htmlelement("a")}} 元素的 `href` 属性一样。
+要想在网页上放置简单的图像，我们需要使用 {{htmlelement("img")}} 元素。这个元素是 {{Glossary("void element, 空元素")}}（即无法包含任何子内容和结束标签），它需要两个属性才能起作用：`src` 和 `alt`。`src` 属性包含一个 URL，该 URL 指向要嵌入页面的图像。`src` 属性可以是相对 URL 或绝对 URL，这与 {{htmlelement("a")}} 元素的 `href` 属性类似。如果没有 `src` 属性，`img` 元素就没有图像可加载。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:42]:
```suggestion
要想在网页上放置简单的图像，我们需要使用 {{htmlelement("img")}} 元素。这个元素是{{Glossary("void element", "空元素")}}（即无法包含任何子内容和结束标签），它需要两个属性才能起作用：`src` 和 `alt`。`src` 属性包含一个 URL，该 URL 指向要嵌入页面的图像。`src` 属性可以是相对 URL 或绝对 URL，这与 {{htmlelement("a")}} 元素的 `href` 属性类似。如果没有 `src` 属性，`img` 元素就没有图像可加载。
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 3]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
-在这个动手练习的部分中，我们希望你把本章节中的上一个动手练习完成的代码拿过来，把它转换为一个 figure:
+在这个动手练习的部分中，我们希望你把上一个动手练习完成的代码拿过来，把它转换为一个 figure:
 
-- 把之前的代码放入 {{htmlelement("figure")}} 元素中。
-- 将 `title` 属性的文本复制出来，删除 `title` 元素，然后把文字放入 {{htmlelement("figcaption")}} 元素中，当然这个元素在 {{htmlelement("img")}} 的下面。
+1. 把之前的代码放入 {{htmlelement("figure")}} 元素中。
+2. 将 `title` 属性的文本复制出来，移除 `title` 元素，然后把文字放入 {{htmlelement("figcaption")}} 元素中，这个元素在 {{htmlelement("img")}} 的下面。
 
-如果你遇到了错误，你可以按 reset 按钮来重置。如果你遇到了困难无法完成，按下 Show solution 按钮来看一下答案。
+如果你遇到了错误，可以按 _reset_ 按钮来重置。如果你遇到了困难无法完成，按下 _Show solution_ 按钮来看一下答案。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:455]:
```suggestion
如果你遇到了错误，可以按*重置*按钮来重置。如果你遇到了困难无法完成，按下*显示答案*按钮来看一下答案。
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 4]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
 
 ### 动手练习：创建一个 figure
 
-在这个动手练习的部分中，我们希望你把本章节中的上一个动手练习完成的代码拿过来，把它转换为一个 figure:
+在这个动手练习的部分中，我们希望你把上一个动手练习完成的代码拿过来，把它转换为一个 figure:
 
-- 把之前的代码放入 {{htmlelement("figure")}} 元素中。
-- 将 `title` 属性的文本复制出来，删除 `title` 元素，然后把文字放入 {{htmlelement("figcaption")}} 元素中，当然这个元素在 {{htmlelement("img")}} 的下面。
+1. 把之前的代码放入 {{htmlelement("figure")}} 元素中。
+2. 将 `title` 属性的文本复制出来，移除 `title` 元素，然后把文字放入 {{htmlelement("figcaption")}} 元素中，这个元素在 {{htmlelement("img")}} 的下面。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:453]:
```suggestion
2. 将 `title` 属性的文本复制出来，移除 `title` 属性，然后把文字放入 {{htmlelement("figcaption")}} 元素中，这个元素在 {{htmlelement("img")}} 的下面。
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 5]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
+  textarea.selectionEnd = caretPos;
+  textarea.focus();
+  textarea.scrollTop = scrollPos;
+}
+
+// 每当用户更新文本区域的代码时，更新保存的 userCode
+
+textarea.onkeyup = () => {
+  // 我们只希望在显示用户代码时保存状态，
+  // 而不保存解决方案，以防止解决方案覆盖用户代码
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:571]:
```suggestion
  // 而不保存答案，以防止答案覆盖用户代码
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 6]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
+  updateCode();
+};
+```
+
+{{ EmbedLiveSample('动手练习：创建一个 figure', 700, 350) }}
 
 ## CSS 背景图片
 
-你也可以使用 CSS 把图片嵌入网站中（JavaScript 也行，不过那是另外一个故事了），这个 CSS 属性 {{cssxref("background-image")}} 和另其他 `background-*` 属性是用来放置背景图片的。比如，为页面中的所有段落设置一个背景图片，你可以这样做：
+你也可以使用 CSS 把图片嵌入网站中（JavaScript 也行，不过那是另外一个故事了）。CSS 属性 {{cssxref("background-image")}} 和其他的 `background-*` 属性是用来控制背景图片的。比如，要想为页面中的每个段落设置一个背景图片，你可以这样做：
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:586]:
```suggestion
你也可以使用 CSS 把图片嵌入网站中（JavaScript 也行，不过那是另外一回事了）。CSS 属性 {{cssxref("background-image")}} 和其他的 `background-*` 属性是用来控制背景图片的。比如，要想为页面中的每个段落设置一个背景图片，你可以这样做：
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 7]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
 ```css
 p {
   background-image: url("images/dinosaur.jpg");
 }
 ```
 
-按理说，这种做法相对于 HTML 中插入图片的做法，可以更好地控制图片和设置图片的位置，那么为什么我们还要使用 HTML 图片呢？如上所述，CSS 背景图片只是为了装饰 — 如果你只是想要在你的页面上添加一些漂亮的东西，来提升视觉效果，那 CSS 的做法是可以的。但是这样插入的图片完全没有语义上的意义，它们不能有任何备选文本，也不能被屏幕阅读器识别。这就是 HTML 图片有用的地方了。
+按理说，这种做法相对于 HTML 中插入图片的做法，可以更好地控制图片和设置图片的位置，那么为什么我们还要使用 HTML 图片呢？如上所述，CSS 背景图片只是为了装饰——如果你只是想要在你的页面上添加一些漂亮的东西，来提升视觉效果，那 CSS 的做法是可以的。但是这样插入的图片完全没有语义上的意义，它们不能有任何备选文本，也不能被屏幕阅读器识别，等等。这就是 HTML 图片有用的地方了。
 
 总而言之，如果图像对你的内容里有意义，则应使用 HTML 图像。如果图像纯粹是装饰，则应使用 CSS 背景图片。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:596]:
```suggestion
总而言之，如果图像对你的内容有意义，则应使用 HTML 图像。如果图像纯粹是装饰，则应使用 CSS 背景图片。
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 8]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
 
-> **备注：** 你可以在 [CSS](/zh-CN/docs/Learn/CSS) 模块里学到更多关于[CSS 背景图片](/zh-CN/docs/Learn/CSS/Styling_boxes/背景)的东西。
+> **备注：** 你可以在 [CSS](/zh-CN/docs/Learn/CSS) 模块里学到更多关于 [CSS 背景图片](/zh-CN/docs/Learn/CSS/Styling_boxes/背景)的知识。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:598]:
```suggestion
> **备注：** 你可以在 [CSS](/zh-CN/docs/Learn/CSS) 模块里学到更多关于 [CSS 背景图片](/zh-CN/docs/Learn/CSS/Building_blocks/Backgrounds_and_borders)的知识。
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 9]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
+
+## 网页上的其他图像
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:600]:
```suggestion
## 网页上的其他图形
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 10]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
+我们已经了解到可以使用 {{HTMLElement("img")}} 元素显示静态图像，或者通过使用 {{cssxref("background-image")}} 属性来设置 HTML 元素的背景。你还可以动态生成图形，或在生成后对图像进行操作。浏览器提供了使用代码创建 2D 和 3D 图形的方法，以及包含来自上传文件或用户摄像头实时流的视频。以下是有关这些更高级图像主题的文章链接：
+
+- [Canvas](/zh-CN/docs/Web/API/Canvas_API)
+  - : {{HTMLElement("canvas")}} 元素提供了使用 JavaScript 绘制 2D 图形的 API。
+- [SVG](/zh-CN/docs/Web/SVG)
+  - : 你可以借助可缩放矢量图形（SVG）来使用线条、曲线和其他几何形状来渲染 2D 图形。借助矢量图形，你可以创建能够以任意尺寸清晰缩放的图像。
+- [WebGL](/zh-CN/docs/Web/API/WebGL_API)
+  - : WebGL API 指南将帮助你入门 WebGL，这是用于 Web 的 3D 图形 API，可让你在 Web 内容中使用标准的 OpenGL ES。
+- [使用 HTML 音频和视频](/zh-CN/docs/Learn/HTML/Multimedia_and_embedding/Video_and_audio_content)
+  - : 与 `<img>` 类似，你可以使用 HTML 将{{htmlelement("video")}} 和 {{htmlelement("audio")}} 嵌入到网页中，并控制其播放。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:611]:
```suggestion
  - : 与 `<img>` 类似，你可以使用 HTML 将 {{htmlelement("video")}} 和 {{htmlelement("audio")}} 嵌入到网页中，并控制其播放。
```

File: files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md [hunk 11]
```diff
@@ -312,194 +412,212 @@ https://raw.githubusercontent.com/mdn/learning-area/master/html/multimedia-and-e
...
+- [WebRTC](/zh-CN/docs/Web/API/WebRTC_API)
+  - : WebRTC 中的 RTC 代表实时通信（Real-Time Communications），这是一种可以在浏览器客户端（对等方）之间进行音频/视频流和数据共享的技术。
+
+## 测试你的技能！
+
+你已经阅读完本文，但你能记住重要的信息吗？在继续之前，你可以进行一些进一步的测试，以验证你是否掌握了这些信息。请参阅[测试你的技能：HTML 图像](/zh-CN/docs/Learn/HTML/Multimedia_and_embedding/Images_in_HTML/Test_your_skills:_HTML_images)。
```
Comment (jasonren0403) [inline files/zh-cn/learn/html/multimedia_and_embedding/images_in_html/index.md:617]: https://github.com/mdn/translated-content/blob/main/docs/zh-cn/glossary.md#标题表格

```suggestion
## 技能测试！

你已经阅读完本文，但你能记住重要的信息吗？在继续之前，你可以进行一些进一步的测试，以验证你是否掌握了这些信息。请参阅[技能测试：HTML 图像](/zh-CN/docs/Learn/HTML/Multimedia_and_embedding/Images_in_HTML/Test_your_skills:_HTML_images)。
```

**Review by jasonren0403 — CHANGES_REQUESTED** [review]
一些修改建议

**Review by jasonren0403 — APPROVED** [review]
👍 Thank you

**Pull request merged**